pkg/skaffold/build/cache/lookup.go

```diff
@@ -59,7 +61,10 @@ func (c *cache) lookup(ctx context.Context, a *latest.Artifact, tag string) cach
 
 	entry, cacheHit := c.artifactCache[hash]
 	if !cacheHit {
-		return needsBuilding{hash: hash}
+		entry, err = c.tryImport(ctx, a, tag, hash)
+		if err != nil {
+			return needsBuilding{hash: hash}
```

Comment:
It would be nice to log the error that caused it to still need building.

Otherwise, PR seems reasonable to me. Anything required to remove the 'Draft' label?

Reply:
Nothing (aside from docs changes), as long as you're happy to make this the default behaviour, as opposed to opt in. I'll have a scan through the docs and see what needs updating.